### ISSUE TITLE
False warning about omitting type

### ISSUE THREAD

**Issue by dsnet** [golang/lint#229]
Consider the following:
```go
var rlast io.ReadSeeker = bytes.NewReader([]byte("foo"))
```

Currently, I get the warning:
```
main.go:9:12: should omit type io.ReadSeeker from declaration of var rlast; it will be inferred from the right-hand side
```

I believe this warning should only be shown if the type on the left is exactly the same as the type on the right. In this case, the left is an `io.ReadSeeker`, while the right side is an `bytes.Reader`.

**Comment by dsnet** [comment]
Fixed in 48c8655188557fb1ac819edd02d3694237eff1ad
